modules/apps/headless/content-space/content-space-headless-apio-impl/src/main/java/com/liferay/content/space/headless/apio/internal/architect/resource/ContentSpaceActionRouterImpl.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.space.headless.apio.internal.architect.resource;
+
+import com.liferay.apio.architect.pagination.PageItems;
+import com.liferay.apio.architect.pagination.Pagination;
+import com.liferay.apio.architect.router.ActionRouter;
+import com.liferay.content.space.headless.apio.architect.model.ContentSpace;
+import com.liferay.content.space.headless.apio.architect.resource.ContentSpaceActionRouter;
+import com.liferay.content.space.headless.apio.architect.util.ContentSpaceUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.comparator.GroupIdComparator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Provides the information necessary to expose content space resources through
+ * a web API. The resources are mapped from the internal model {@code Group}.
+ *
+ * @author Javier Gamarra
+ * @author Cristina González
+ */
+@Component(immediate = true, service = ActionRouter.class)
+public class ContentSpaceActionRouterImpl
+	implements ContentSpaceActionRouter, ActionRouter<ContentSpace> {
```

Comment:
Although it's going to be generated I think that ContentSpaceActionRouter should extend ActionRouter<ContentSpace>